### PR TITLE
[XPU] Fix UT errors in the sam3 and lfm series model.

### PR DESCRIPTION
**Pull request by YangKai0616**
# What does this PR do?

This PR:

1. Update the ground truth of `lfm_moe` to make it pass on XPU. (My local testing indicates that CUDA also requires the update)
2. Add deterministic algorithms to the `sam3` test on XPU (Correctness has higher priority. Our team will subsequently address determinism).
3. XPU's FA2 now supports local mask, remove the previous comment.

The test cases corresponding to each point are as follows:

1.
```
tests/models/lfm2_moe/test_modeling_lfm2_moe.py::Lfm2MoeIntegrationTest::test_model_1a8b_batched_chat_generation
tests/models/lfm2_moe/test_modeling_lfm2_moe.py::Lfm2MoeIntegrationTest::test_model_1a8b_logits
```
2.
```
tests/models/sam3/test_modeling_sam3.py::Sam3ModelIntegrationTest::test_efficient_multi_prompt_single_image
tests/models/sam3/test_modeling_sam3.py::Sam3ModelIntegrationTest::test_efficient_single_prompt_multi_images
```
3.
```
tests/models/afmoe/test_modeling_afmoe.py::AfmoeModelTest::test_flash_attn_2_inference_equivalence_right_padding
```

Additionally, regarding the example `tests/models/sam3_tracker_video/test_modeling_sam3_tracker_video.py::Sam3TrackerVideoModelIntegrationTest::test_inference_mask_generation_video_batched_bb`, the results I obtained on XPU are
```
[[[[-17.2589, -17.2589], [-17.2777, -17.2777]]], [[[-8.5523, -8.5523], [-8.5103, -8.5103]]]],
[[[[-17.8108, -17.8108], [-18.1583, -18.1583]]], [[[-9.1217, -9.1217], [-9.2395, -9.2395]]]],
[[[[-17.9431, -17.9431], [-18.4636, -18.4636]]], [[[-10.8989, -10.8989], [-11.0142, -11.0142]]]]
```
, which are quite close to those obtained with CUDA. The CUDA's results are
```
[[[[-17.2589, -17.2589], [-17.2777, -17.2777]]], [[[-8.5523, -8.5523], [-8.5103, -8.5103]]]],
[[[[-17.8107, -17.8107], [-18.1581, -18.1581]]], [[[-9.1150, -9.1150], [-9.2327, -9.2327]]]],
[[[[-17.9432, -17.9432], [-18.4637, -18.4637]]], [[[-10.9026, -10.9026], [-11.0184, -11.0184]]]]
```
Considering the platform differences, can we increase the `rtol` to allow XPU to pass the test? If there are any issues, please feel free to contact me!

### CODE REVIEW

**Comment by YangKai0616** [comment]
@ydshieh ,  please help review, thanks!

**Review by ydshieh — APPROVED** [review]
Thank you a lot @YangKai0616 .

I will check the cuda part then merge the PR.
